qiskit/circuit/barrier.py

```diff
@@ -21,7 +21,10 @@
 
 
 class Barrier(Instruction):
-    """Barrier instruction."""
+    """A barrier is a visual indicator of the grouping of a circuit section. 
+    Actings as a directive for circuit compilation to separate pieces 
+    of a circuit so that any optimizations or re-writes are constrained 
+    to only act between barriers."""
```

Comment:
In sphinx/docstrings the first line has special meaning it's basically a summary line that is shown in table of contents and other places. So in general it's better to have a consise top line like what was there before. I do like this description but can you do something like:

```
"""
Barrier instruction

A barrier is ...